uaclient/fix.py

```diff
@@ -0,0 +1,720 @@
+        Field("binary_package_version", StringDataValue),
+        Field("source_package", StringDataValue),
+    ]
+
+    def __init__(
+        self,
+        *,
+        binary_package: str,
+        binary_package_version: str,
+        source_package: str
+    ):
+        self.source_package = source_package
+        self.binary_package = binary_package
+        self.binary_package_version = binary_package_version
+
+
+class FixPlanWarningPackageCannotBeInstalled(FixPlanWarning):
+    fields = [
+        Field("warning_type", StringDataValue),
+        Field("order", IntDataValue),
+        Field("data", SecurityIssueNotFixedData),
+    ]
+
+    def __init__(self, *, order: int, data: PackageCannotBeInstalledData):
+        super().__init__(
+            warning_type=FixWarningType.PACKAGE_CANNOT_BE_INSTALLED.value,
+            order=order,
+        )
+        self.data = data
+
+
+class FixPlanError(DataObject):
+    fields = [
+        Field("msg", StringDataValue),
+        Field("code", StringDataValue, required=False),
+    ]
+
+    def __init__(self, *, msg: str, code: Optional[str]):
+        self.msg = msg
+        self.code = code
+
+
+class FixPlanResult(DataObject):
+    fields = [
+        Field("title", StringDataValue),
+        Field("expected_status", StringDataValue),
+        Field("plan", data_list(FixPlanStep)),
+        Field("warnings", data_list(FixPlanWarning), required=False),
+        Field("error", FixPlanError, required=False),
+    ]
+
+    def __init__(
+        self,
+        *,
+        title: str,
+        expected_status: str,
+        plan: List[FixPlanStep],
+        warnings: List[FixPlanWarning],
+        error: Optional[FixPlanError]
+    ):
+        self.title = title
+        self.expected_status = expected_status
+        self.plan = plan
+        self.warnings = warnings
+        self.error = error
+
+
+class FixPlanUSNResult(DataObject):
+    fields = [
+        Field("target_usn_plan", FixPlanResult),
+        Field("related_usns_plan", data_list(FixPlanResult), required=False),
+    ]
+
+    def __init__(
+        self,
+        *,
+        target_usn_plan: FixPlanResult,
+        related_usns_plan: List[FixPlanResult]
+    ):
+        self.target_usn_plan = target_usn_plan
+        self.related_usns_plan = related_usns_plan
+
+
+class FixPlan:
+    def __init__(self, title: str):
+        self.order = 1
+        self.title = title
+        self.fix_steps = []  # type: List[FixPlanStep]
+        self.fix_warnings = []  # type: List[FixPlanWarning]
+        self.error = None  # type: Optional[FixPlanError]
+
+    def register_step(
+        self,
+        operation: FixStepType,
+        data: Dict[str, Any],
+    ):
+        # just to make mypy happy
+        fix_step = None  # type: Optional[FixPlanStep]
+
+        if operation == FixStepType.ATTACH:
+            fix_step = FixPlanAttachStep(
+                order=self.order, data=AttachData.from_dict(data)
+            )
```

Comment:
I don't remember the `order` field or why we need it.
It looks like keeping track of `order` complicates the `register_step` (and `register_warning` etc) functions because without it I think registering a step could look more like
```
plan.register_step(FixPlanNoOpStep(data=NoOpData(status=FixPlanNoOpStatus.NOT_AFFECTED)))
```

But if we do need `order`, then this looks good

Reply:
The order is just to for the user to easily combine what we have in warning with the plan and get the exact order of execution of fix. Additinally, we can use that when we use this endpoint in the CLI, allowing us to keep the same message order we currently have